shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -306,6 +310,13 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
     flutterArguments.aot_data = _aotData.get();
   }
 
+  // Only create a Compositor if we have a ViewController.
+  if (_viewController) {
+    FlutterCompositor compositor = {};
+    [self setupCompositor:&compositor];
```

Comment:
why not pass `&flutterArguments.compositor` here? Also won't compositor go out of scope here?

Reply:
I think the compositor gets copied in InferExternalViewEmbedderFromArgs before it goes out of scope.

Passing `flutterArguments.compositor` directly gives an error right now
```
cannot initialize a parameter of type 'FlutterCompositor *' with an lvalue of type 'const FlutterCompositor *'
```